File: src/ros_topics_services/src/talker_listener/scripts/listener.py
#!/user/bin/env python

import rospy
from std_msgs.msg import String

def callback_function(message):
    rospy.loginfo(rospy.get_caller_id() +" I heard %s", message.data)

def listener():
    rospy.init_node('listener_node', anonymous=True)
    sub = rospy.Subscriber('talker_topic',String,callback_function)
    rospy.spin()

if __name__== '__main__':
    try:
        listener()
    except rospy.ROSInterruptException:
        pass

File: src/ros_topics_services/src/talker_listener/listener.cpp
#include "ros/ros.h"
#include "std_msgs/String.h"


void callback_function(const std_msgs::String::ConstPtr& msg)
{
    ROS_INFO("[Listener] I heard: [%s] \n", msg->data.c_str());
}

int main(int argc,char **argv)
{
    ros::init(argc, argv, "listener_node");
    ros::NodeHandle node;
    ros::Subscriber sub = node.subscribe("talker_topic", 100, callback_function);
    ros::spin();

    return 0;
}